daemon/policy.go

```diff
@@ -98,6 +98,43 @@ func (d *Daemon) TriggerPolicyUpdates(added []policy.NumericIdentity) {
 	d.endpointsMU.RUnlock()
 }
 
+func (d *Daemon) UpdatePolicyEnforcement(e *endpoint.Endpoint) bool {
+	if d.conf.EnablePolicy == endpoint.AlwaysEnforce {
+		return true
+	} else if d.conf.EnablePolicy == endpoint.DefaultEnforcement && !d.conf.IsK8sEnabled() {
+		log.Infof("updatePolicyEnforcement: default enforcement, no k8s")
+		if d.GetPolicyRepository().NumRules() > 0 {
+			log.Infof("updatePolicyEnforcement: set to true, num rules > 0 ")
+			// TODO - revisit setting Daemon endpoint.OptionPolicy here
+			d.conf.Opts.Set(endpoint.OptionPolicy, true)
+			return true
+		} else {
+			log.Infof("updatePolicyEnforcement: set to false, num rules == 0")
+			d.conf.Opts.Set(endpoint.OptionPolicy, false)
+			return false
+		}
+	} else if d.conf.EnablePolicy == endpoint.DefaultEnforcement && d.conf.IsK8sEnabled() {
+		e.Mutex.RLock()
+		// Convert to LabelArray so we can pass to Matches function later.
+		var endpointLabels labels.LabelArray
+		for _, lbl := range e.Consumable.LabelList {
+			endpointLabels = append(endpointLabels, lbl)
+		}
+		e.Mutex.RUnlock()
+		d.GetPolicyRepository().Mutex.RLock()
+		defer d.GetPolicyRepository().Mutex.RUnlock()
+		// Check if rules match the labels for this endpoint.
+		// If so, enable policy enforcement.
+		rulesMatch := d.GetPolicyRepository().GetRulesMatching(endpointLabels)
```

Comment:
you can simply `return d.GetPolicyRepository().GetRulesMatching(endpointLabels)`